pkg/streamformatter/streamformatter_test.go

```diff
@@ -84,9 +85,25 @@ func TestJSONFormatProgress(t *testing.T) {
 	if msg.Status != "action" {
 		t.Fatalf("Status must be 'action', got: %s", msg.Status)
 	}
-	if msg.ProgressMessage != progress.String() {
-		t.Fatalf("ProgressMessage must be %s, got: %s", progress.String(), msg.ProgressMessage)
+
+	// The progress will always be in the format of:
+	// [=========================>                         ]     15 B/30 B 404933h7m11s
+	// The last entry '404933h7m11s' is the timeLeftBox.
+	// However, the timeLeftBox field may change as progress.String() depends on time.Now().
+	// Therefore, we have to strip the timeLeftBox from the strings to do the comparison.
+
+	// Save the progress.String()
+	progressString := progress.String()
+
+	// Find the index before the timeLeftBox
+	stringIndex := strings.LastIndex(progressString, " ")
+	messageIndex := strings.LastIndex(msg.ProgressMessage, " ")
+
+	// Compare the progress strings before the timeLeftBox
+	if msg.ProgressMessage[:messageIndex] != progressString[:stringIndex] {
+		t.Fatalf("ProgressMessage without the timeLeftBox must be %s, got: %s", progressString[:stringIndex], msg.ProgressMessage[:messageIndex])
```

Comment:
Yes I think compare against 
```"[=========================>                         ]     15 B/30 B"```
will do it as the current (15) and total (30) values of the progress are fixed. Let me update the PR.